daemon/containerd/image.go

```diff
@@ -158,15 +163,18 @@ func (i *ImageService) GetImageManifest(ctx context.Context, refOrID string, opt
 	}
 
 	if containerdimages.IsManifestType(desc.MediaType) {
+		if options.Platform != nil {
+			if desc.Platform == nil {
+				return nil, errdefs.NotFound(errors.Errorf("image with reference %s was found but does not match the specified platform: wanted %s, actual: nil", refOrID, cplatforms.Format(*options.Platform)))
```

Comment:
would the same apply here? i.e. if desc doesn't have a Platform, that would be "any" platform? (so no error?)

Reply:
So actually the test I'm trying to fix has an image with no platform... 

The GD code creates a platform no matter what: 

* https://github.com/moby/moby/blob/54d84229ce50ae1e0d9af6bb4410be629d2df30b/daemon/images/image.go#L207-L211